threestudio/systems/gaussian_splatting.py

```diff
@@ -0,0 +1,256 @@
+import os
+import math
+from dataclasses import dataclass
+
+import torch
+import numpy as np
+
+import threestudio
+from threestudio.systems.base import BaseLift3DSystem
+from threestudio.models.geometry.gaussian import BasicPointCloud
+from threestudio.utils.typing import *
+
+def getWorld2View2(R, t, translate=np.array([.0, .0, .0]), scale=1.0):
+    Rt = np.zeros((4, 4))
+    Rt[:3, :3] = R.transpose()
+    Rt[:3, 3] = t
+    Rt[3, 3] = 1.0
+
+    C2W = np.linalg.inv(Rt)
+    cam_center = C2W[:3, 3]
+    cam_center = (cam_center + translate) * scale
+    C2W[:3, 3] = cam_center
+    Rt = np.linalg.inv(C2W)
+    return np.float32(Rt)
+
+def getProjectionMatrix(znear, zfar, fovX, fovY):
+    tanHalfFovY = math.tan((fovY / 2))
+    tanHalfFovX = math.tan((fovX / 2))
+
+    top = tanHalfFovY * znear
+    bottom = -top
+    right = tanHalfFovX * znear
+    left = -right
+
+    P = torch.zeros(4, 4)
+
+    z_sign = 1.0
+
+    P[0, 0] = 2.0 * znear / (right - left)
+    P[1, 1] = 2.0 * znear / (top - bottom)
+    P[0, 2] = (right + left) / (right - left)
+    P[1, 2] = (top + bottom) / (top - bottom)
+    P[3, 2] = z_sign
+    P[2, 2] = z_sign * zfar / (zfar - znear)
+    P[2, 3] = -(zfar * znear) / (zfar - znear)
+    return P
+
+def get_cam_info(c2w, fovy):
+    matrix = np.linalg.inv(c2w[0].cpu().numpy())
+    R = np.transpose(matrix[:3,:3])
+    R[:,0] = -R[:,0]
+    T = -matrix[:3, 3]
+    
+    world_view_transform = torch.tensor(getWorld2View2(R, T)).transpose(0, 1).cuda()
+    projection_matrix = getProjectionMatrix(znear=0.01, zfar=100, fovX=fovy, fovY=fovy).transpose(0,1).cuda()
+    full_proj_transform = (world_view_transform.unsqueeze(0).bmm(projection_matrix.unsqueeze(0))).squeeze(0)
+    camera_center = world_view_transform.inverse()[3, :3]
+    
+    return world_view_transform, full_proj_transform, camera_center
+
+class Camera(NamedTuple):
+    FoVx: torch.Tensor
+    FoVy: torch.Tensor
+    camera_center: torch.Tensor
+    image_width: int
+    image_height: int
+    world_view_transform: torch.Tensor
+    full_proj_transform: torch.Tensor
+        
```

Comment:
What do you think about moving these camera transformation functions to the utils folder? It could make our codebase more organized.